e2e/nomostest/registryproviders/helm.go

```diff
@@ -0,0 +1,142 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package registryproviders
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"kpt.dev/configsync/e2e/nomostest/gitproviders"
+	"kpt.dev/configsync/e2e/nomostest/testshell"
+	"sigs.k8s.io/kustomize/kyaml/copyutil"
+	"sigs.k8s.io/yaml"
+)
+
+// use an auto-incrementing index to create unique file names for tarballs
+var helmIndex int
+
+// BuildHelmPackage creates a new OCIImage object and associated tarball using the provided
+// Repository. The contents of the git repository will be bundled into a tarball
```

Comment:
Do we need to do this in the git repo? Can we use a separate directory so we don't include the .git meta files?

Reply:
Or does it need to be tied to a git repo for some testing reasons? Seems kinda tightly coupled, but maybe this code predates this PR?

Reply:
This design was deliberate with the intent of minimizing the number of primitives that the tests need to work with. The intent is basically that you can add files to the `Repository` object and then push the artifact to any sync source (git, oci, helm, etc). This way we can also deal with versioning consistently (e.g. tag things with the git branch).

I think this works pretty well for oci from the previous PR, but admittedly feels a bit clunkier with helm because the file/directory structure is very specific to helm.